2010-02-27  Aurelien Jarno  <aurelien@aurel32.net>

	* pthreadP.h(INVALID_TD_P, INVALID_NOT_TERMINATED_TD_P): detect
	NULL pointers.

---
 nptl/pthreadP.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/nptl/pthreadP.h
+++ b/nptl/pthreadP.h
@@ -206,8 +206,8 @@
 /* Simplified test.  This will not catch all invalid descriptors but
    is better than nothing.  And if the test triggers the thread
    descriptor is guaranteed to be invalid.  */
-# define INVALID_TD_P(pd) __builtin_expect ((pd)->tid <= 0, 0)
-# define INVALID_NOT_TERMINATED_TD_P(pd) __builtin_expect ((pd)->tid < 0, 0)
+# define INVALID_TD_P(pd) __builtin_expect (!pd || ((pd)->tid <= 0), 0)
+# define INVALID_NOT_TERMINATED_TD_P(pd) __builtin_expect (!pd || ((pd)->tid < 0), 0)
 #endif
 
 
